x-pack/metricbeat/module/azure/billing/data.go

```diff
@@ -37,81 +43,78 @@ func EventsMapping(subscriptionId string, results Usage, timeOpts TimeIntervalOp
 				"cloud.provider": "azure",
 			}
 
-			if legacy, isLegacy := ud.AsLegacyUsageDetail(); isLegacy {
-
+			switch usageDetails := ud.(type) {
+			case *armconsumption.LegacyUsageDetail:
 				//
 				// legacy data format
 				//
 
+				legacy := usageDetails
+
 				event.ModuleFields = mapstr.M{
-					"subscription_id":   legacy.SubscriptionID,
-					"subscription_name": legacy.SubscriptionName,
+					"subscription_id":   legacy.Properties.SubscriptionID,
+					"subscription_name": legacy.Properties.SubscriptionName,
 					"resource": mapstr.M{
-						"name":  legacy.ResourceName,
-						"type":  legacy.ConsumedService,
-						"group": legacy.ResourceGroup,
+						"name":  legacy.Properties.ResourceName,
+						"type":  legacy.Properties.ConsumedService,
+						"group": legacy.Properties.ResourceGroup,
 					},
 				}
 				event.MetricSetFields = mapstr.M{
 					// original fields
 					"billing_period_id": legacy.ID,
-					"product":           legacy.Product,
-					"pretax_cost":       legacy.Cost,
-					"currency":          legacy.BillingCurrency,
-					"department_name":   legacy.InvoiceSection,
-					"account_name":      legacy.BillingAccountName,
+					"product":           legacy.Properties.Product,
+					"pretax_cost":       strconv.FormatFloat(*legacy.Properties.Cost, 'f', -1, 64),
```

Comment:
Oh, I missed this in the first pass: they moved away from `*decimal.Decimal`!

Why are we converting the value from `float64` to `string`?

Reply:
Got it.

I see those field mappings are defined as a `float`. For example, here's the mapping for the `pretax_cost` field:

```YAML
    - name: pretax_cost
      type: float
      description: >
        The amount of cost before tax.
```

I understand these `string` values are then converted to `float` in Elasticsearch. 

Maybe we can avoid converting `float` > `string` > `float` sticking to `float` yields the same result. 

TBH I don't have enough experience here to say it's equivalent. Let's test our assumptions or bring someone else to this conversation.

WDYT?